docs/validator.go

```diff
@@ -0,0 +1,164 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/a8m/mark"
+	"github.com/minio/cli"
+)
+
+var templateMain = `
+// +build ignore
+
+package main
+
+import (
+    "fmt"
+
+    "github.com/minio/minio-go"
+)
+
+func main() {
+        // Use a secure connection.
+        ssl := true
+
+        // Initialize minio client object.
+        minioClient, err := minio.New("play.minio.io:9000", "Q3AM3UQ867SPQQA43P2F", "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG", ssl)
+        if err != nil {
+                fmt.Println(err)
+                return
+        }
+
+        {{.Text}}
+
+}`
+
+func init() {
+	// Validate go binary.
+	if _, err := exec.LookPath("go"); err != nil {
+		panic(err)
+	}
+}
+
+var globalFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "m",
+		Value: "API.md",
+		Usage: "Path to API.md",
+	},
+	cli.IntFlag{
+		Name:  "skip",
+		Value: 2,
+		Usage: "Skip entries before validating the code.",
+	},
+}
+
+func validatorAction(ctx *cli.Context) error {
+	if !ctx.IsSet("m") {
+		return nil
+	}
+	docPath := ctx.String("m")
+	data, err := ioutil.ReadFile(docPath)
+	if err != nil {
+		return err
+	}
+
+	skipEntries := ctx.Int("skip")
+	m := mark.New(string(data), &mark.Options{
+		Gfm: true, // Github markdown support is enabled by default.
+	})
+
+	t := template.Must(template.New("examples").Parse(templateMain))
+	tmpDir, err := ioutil.TempDir("", "md-verifier")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	fmt.Println("About to run validation tests in", tmpDir)
+	entryN := 1
+	for i := mark.NodeText; i < mark.NodeCheckbox; i++ {
+		if mark.NodeCode != mark.NodeType(i) {
+			m.AddRenderFn(mark.NodeType(i), func(node mark.Node) (s string) {
+				return ""
+			})
+			continue
+		}
+		m.AddRenderFn(mark.NodeCode, func(node mark.Node) (s string) {
+			p, ok := node.(*mark.CodeNode)
+			if !ok {
+				return
+			}
+			p.Text = strings.NewReplacer("&lt;", "<", "&gt;", ">", "&quot;", `"`, "&amp;", "&").Replace(p.Text)
+			if skipEntries > 0 {
+				skipEntries--
+				return
+			}
+
+			testFilePath := filepath.Join(tmpDir, fmt.Sprintf("%d.go", entryN))
+			testBinPath := filepath.Join(tmpDir, fmt.Sprintf("%d", entryN))
+			w, werr := os.Create(testFilePath)
+			if werr != nil {
+				panic(werr)
+			}
+			t.Execute(w, p)
+			w.Sync()
+			w.Close()
+			entryN++
+
+			cmdArgs := []string{"-s", "-w", "-l", testFilePath}
+			exec.Command("gofmt", cmdArgs...).Run()
+
+			cmdArgs = []string{"-w", testFilePath}
+			exec.Command("goimports", cmdArgs...).Run()
```

Comment:
validation fails for `FGetEncryptedObject`, looks like `encrypt` is not getting imported

```sh
Failed to validate code at docs/API.md:18247

// Generate a master symmetric key
key := encrypt.NewSymmetricKey([]byte("my-secret-key-00"))

// Build the CBC encryption material
cbcMaterials, err := encrypt.NewCBCSecureMaterials(key)
if err != nil {
    fmt.Println(err)
    return
}

err = minioClient.FGetEncryptedObject("mybucket", "myobject", "/tmp/myobject", cbcMaterials)
if err != nil {
    fmt.Println(err)
    return
}

```

Reply:
It runs fine on CI looks like you have a local issue `goimports` has a bug btw it stops scanning imports if it reaches a directory which it cannot access.

Reply:
> goimports has a bug btw it stops scanning imports if it reaches a directory which it cannot access.

This is what I faced. 

> Go to the directory and check manually what is going on.

will do